### DISCUSSION TITLE
I don't see the x-robots-tag on some pages.

### DISCUSSION THREAD

**Discussion by limjimwork** [vercel/community#3223] [Help]
### Summary

I am using a custom domain for my staging environment.
I applied the method provided by the following vercel page,
https://vercel.com/guides/are-vercel-preview-deployment-indexed-by-search-engines#x-robots-tag-header,
I set `x-robots-tag: noindex`, but when I do `curl - I` don't see the `x-robots-tag`.

```
module.exports = {
  async headers() {
    const headers = [];
    if (process.env.NEXT_PUBLIC_VERCEL_ENV === 'preview') {
      headers.push({
        headers: [
          {
            key: 'X-Robots-Tag',
            value: 'noindex',
          },
        ],
        source: '/:path*',
      });
    }
    return headers;
  },
 };
```

And when I checked the site,
I could see the `x-robots-tag` on some pages in my website, but not on the other pages.
But it works on localhost.

### Steps to Reproduce

I deploy with vercel and use the next.js framework.

**Comment by swarnava** [comment]
Hi, Could you share a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) repository that can be shared publicly as a **git repository** so that we can take a look?

**Reply by limjimwork** [reply]
Hello, I apologize for the late response. I solved the issue by registering the problematic domain in the Google Search Console, deleting the existing cache, and requesting to exclude new crawling.

**Comment by myznikovgleb** [comment]
@limjimwork 

**tldr** with your `next.config.js` you can only set `x-robots-tag` to routes like `example.com/*` not `example.com/someroute/*` or any other

I guess you just copy `next.config.js` example provided at the guide above. And this example config is not fit to your particular case. I highly recommend you take a look at [this nextjs docs article](https://nextjs.org/docs/app/api-reference/next-config-js/headers) to get familiar with `next.config.js` structure. It helps me a lot

Also I recommend you to check is `process.env.NEXT_PUBLIC_VERCEL_ENV` variable defined at runtime of `headers()` method

**Reply by limjimwork** [reply]
Hello, I apologize for the late response. I solved the issue by registering the problematic domain in the Google Search Console, deleting the existing cache, and requesting to exclude new crawling.
